types/fixed.go

```diff
@@ -88,7 +89,11 @@ func (s *FixedDefinition) ResolveReferences(n *Namespace) error {
 
 func (s *FixedDefinition) Schema(names map[QualifiedName]interface{}) interface{} {
 	name := s.name.String()
-	names[s.name] = 1
+
+	// Add a small hash suffix to the name to avoid name collisions
```

Comment:
Personally, I wouldn't do it just yet because it's just two places and the logic is so simple (feel free to disagree though).

Just thinking it'd be
```
func hash() string {
  return uuid.NewV4().String()[:4]
}
```